app/src/main/java/org/kiwix/kiwixmobile/nav/destination/reader/ReaderFragment.kt

```diff
@@ -85,13 +146,16 @@ class ReaderFragment : CoreReaderFragment() {
 
   override fun onResume() {
     super.onResume()
-    if (zimReaderContainer.zimFile == null && HOME_URL != getCurrentWebView().url) {
+    if (zimReaderContainer.zimFile == null) {
       showHomePage()
     }
   }
 
   override fun onBackPressed(activity: AppCompatActivity): Super {
-    if (super.onBackPressed(activity) == ShouldCall) {
+    val callType = super.onBackPressed(activity)
+    if (callType == ShouldCall && getCurrentWebView().canGoBack()) {
+      getCurrentWebView().goBack()
+    } else if (callType == ShouldCall) {
       getActivity()?.finish()
```

Comment:
I think returning `ShouldCall` here will just tell the activity to do normal `onBackPressed` stuff which should be finishing the activity?

Reply:
nope, sorry, reading the code better now